rpc/types/utils.go

```diff
@@ -266,8 +271,13 @@ func TxExceedBlockGasLimit(res *abci.ResponseDeliverTx) bool {
 	return strings.Contains(res.Log, ExceedBlockGasLimitError)
 }
 
+// TxCommitError returns true if the tx commit error.
+func TxCommitError(res *abci.ResponseDeliverTx) bool {
+	return strings.Contains(res.Log, CommitError)
+}
+
 // TxSuccessOrExceedsBlockGasLimit returnsrue if the transaction was successful
 // or if it failed with an ExceedBlockGasLimit error
 func TxSuccessOrExceedsBlockGasLimit(res *abci.ResponseDeliverTx) bool {
-	return res.Code == 0 || TxExceedBlockGasLimit(res)
+	return res.Code == 0 || TxExceedBlockGasLimit(res) || TxCommitError(res)
```

Comment:
Would be better to refactor the name of this func to `TxSucess`

Reply:
Has been refactored into a more specific name `TxStateDBCommitError`

Reply:
yes, thanks!
But was referring to the `TxSuccessOrExceedsBlockGasLimit`, cause now there're 3 conditions instead of 2 (success and exceed block gas limit), maybe something like `TxSucessOrExpectedFailure` and then update the comments to explain which are the expected failure cases